### ISSUE TITLE
Implement new Concept Exercise: higher-order-functions

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#864]
This issue describes how to implement the `higher-order-functions` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student more advanced aspects of the Concept of Higher-order Functions in F#.

## Learning objectives

- Know how to define and read function types.
- Know how to return functions.
- Know how to pass functions as arguments.
- Know how to define lambda functions.
- Know of some built-in higher-order functions (e.g. `List.exists`).

## Out of scope

- Recursion.
- Mutually recursive functions.
- Custom operators.
- Monadic functions.
- Function composition.

## Concepts

The Concepts this exercise unlocks are:

- `higher-order-functions`: know how to return functions; know how to pass functions as arguments; know how to define lambda functions; know of some built-in higher-order functions .

## Prerequisites

This exercise's prerequisites Concepts are:

- `lists`: know how to work with lists, which has many built-in higher-order functions we can use.

Any other data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Functions][functions]: introduction to functions in F#.
- [Lambda expressions][lambda-expressions]: how to define a lambda expression.
- [Function values][function-values]: discusses function types and passing functions as values.

### After

- [Functions][functions]: introduction to functions in F#.
- [Lambda expressions][lambda-expressions]: how to define a lambda expression.
- [Function values][function-values]: discusses function types and passing functions as values.
- [First-class functions][first-class-functions]: overview of first-class function features of F#.
- [Defining functions][definining-functions]: more in-depth discussion on how to define functions.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/types/function.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[functions]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/functions/
[first-class-functions]: https://docs.microsoft.com/en-us/dotnet/fsharp/introduction-to-functional-programming/first-class-functions
[definining-functions]: https://fsharpforfunandprofit.com/posts/defining-functions/
[lambda-expressions]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/functions/#lambda-expressions
[function-values]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/functions/#function-values
